### DISCUSSION TITLE
Spanner integration?

### DISCUSSION THREAD

**Discussion by Jonpez2** [ory/keto#581] [Q&A] [answered]
I’ve been digging around in this project as it seems like it might be a good fit for my gcp-based project. I was trying to figure out which backend databases you support; cockroachdb certainly seems like an option but do you support spanner?

thank you!

**Comment by vinckr** [comment, marked as answer]
We are planning to integate the Google Cloud Spanner capabilities into Keto, with cockroachdb as the main supported database, but afaik all solutions following the general PostgreSQL protocol should be supported. 
Also have a look at this issue: https://github.com/ory/keto/issues/504

**Reply by zepatrik** [reply]
As an open source project this cannot primarily rely on closed source services. Adding spanner support later on after the base technology is done will be possible, but at the current stage we have to first settle on a stable and battle tested technology.
